apps/prairielearn/assets/scripts/lib/parallelHistograms.ts

```diff
@@ -0,0 +1,201 @@
+import { scaleLinear, scaleBand, axisBottom, axisLeft, range, select } from 'd3';
+
+interface Data {
+  label: string;
+  histogram: number[];
+  mean: number;
+}
+
+export function parallel_histograms(
+  selector: Element,
+  data: Data[],
+  options: {
+    width?: number;
+    height?: number;
+    xTickLabels?: string[] | 'auto';
+    yTickLabels?: string[] | 'auto';
+    xAxisHeight?: number;
+    yAxisWidth?: number;
+    xgrid?: number[];
+    ygrid?: number[];
+    xlabel?: string;
+    ylabel?: string;
+    topPadding?: number;
+    rightPadding?: number;
+  },
+) {
+  const resolvedOptions = {
+    width: 600,
+    height: 370,
+    xTickLabels: 'auto',
+    yTickLabels: 'auto',
+    yAxisWidth: 70,
+    xAxisHeight: 70,
+    topPadding: 15,
+    rightPadding: 2,
+    ...options,
+  };
+
+  const width = resolvedOptions.width;
+  const height = resolvedOptions.height;
+
+  const yTickLabels = resolvedOptions.yTickLabels;
+
+  const yAxisWidth = resolvedOptions.yAxisWidth;
+  const xAxisHeight = resolvedOptions.xAxisHeight;
+
+  const topPadding = resolvedOptions.topPadding;
+  const rightPadding = resolvedOptions.rightPadding;
+
+  const totalWidth = width + yAxisWidth + rightPadding;
+  const heightWithPadding = height + topPadding;
+  const totalHeight = heightWithPadding + xAxisHeight;
+
+  const numBuckets = data[0].histogram.length;
+  const numHistograms = data.length;
+
+  const yLinear = scaleLinear().domain([0, numBuckets]).range([0, height]);
+
+  const xOrdinal = scaleBand()
+    .domain(range(numHistograms).map((d) => `${d}`))
+    .rangeRound([0, width])
+    .padding(0.0);
+
+  const xLinear = scaleLinear().domain([0, numHistograms]).range([0, width]);
+
+  const plot = select(selector)
+    .insert('svg', ':first-child')
+    .attr('width', totalWidth)
+    .attr('height', totalHeight)
+    .attr('class', 'center-block statsPlot');
+
+  const verticalGridLinear = axisBottom(xLinear)
+    .tickSize(-height)
+    .tickFormat(() => '');
+
+  plot
+    .append('g')
+    .attr('class', 'grid')
+    .attr('transform', 'translate(' + yAxisWidth + ',' + heightWithPadding + ')')
```

Comment:
Backtick template literals? I see Jonatan already brought this up on in `histogram.ts`. I'll leave it to you to find all the remaining places we could make this change.

Reply:
Now using backtick template literals here and other remaining places (I think I got them all?)  in this file in [4dad954](https://github.com/PrairieLearn/PrairieLearn/pull/9809/commits/4dad954294139e1ccb0018548e102663ab43c2df).